csrc/scheduler/utils.cpp

```diff
@@ -1216,6 +1216,24 @@ std::vector<TensorView*> cacheInputs(Fusion* fusion, bool unroll) {
     for (auto use : tv->uses()) {
       if (!use->isA<PadOp>()) {
         cached_uses.push_back(use);
+      } else {
+        // propagate allocation domain on pad op
```

Comment:
I feel it's a little odd that "cache inputs" would mean some consumer tensors got their allocation domains modified. Also, this would not be necessary if the pad op isn't vectorized.

Reply:
> it's a little odd that "cache inputs" would mean some consumer tensors got their allocation domains modified.

Agree on this one. If we are worried about it, I assume I can add another copy afterwards to restore the allocation domain of the unmodified output from PadOp, basically replacing `tv1 = pad(tv0, ...)` with
```
permuted_tv = pad(tv0, ...)
tv1 = set(permuted_tv)
```
This will be similar behavior to cacheAfter. i.e. we are not interfering with explicitly allocation domain on tv1. 

> Also, this would not be necessary if the pad op isn't vectorized.

That's true. I guess I can leave it as-is if no vectorization is applied, since at the time of scheduling, we already have the whole PointwiseParams.

Reply:
To be clear, I'm not worried about modifying the allocation domain of `tv1`. It's just odd that it's done by some function named "cache inputs" since it has nothing to do with input caching.